external/model-preparation-algorithm/mpa_tasks/apis/segmentation/task.py

```diff
@@ -208,45 +209,33 @@ def _init_test_data_cfg(self, dataset: DatasetEntity):
         )
         return data_cfg
 
-    def _add_predictions_to_dataset_item(self, prediction, feature_vector, dataset_item, save_mask_visualization):
-        soft_prediction = np.transpose(prediction, axes=(1, 2, 0))
-        hard_prediction = create_hard_prediction_from_soft_prediction(
-            soft_prediction=soft_prediction,
-            soft_threshold=self._hyperparams.postprocessing.soft_threshold,
-            blur_strength=self._hyperparams.postprocessing.blur_strength,
-        )
-        annotations = create_annotation_from_segmentation_map(
-            hard_prediction=hard_prediction,
-            soft_prediction=soft_prediction,
-            label_map=self._label_dictionary,
-        )
-        dataset_item.append_annotations(annotations=annotations)
-
-        if feature_vector is not None:
-            active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
-            dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
-
-        if save_mask_visualization:
-            for label_index, label in self._label_dictionary.items():
-                if label_index == 0:
-                    continue
-
-                if len(soft_prediction.shape) == 3:
-                    current_label_soft_prediction = soft_prediction[:, :, label_index]
-                else:
-                    current_label_soft_prediction = soft_prediction
-                min_soft_score = np.min(current_label_soft_prediction)
-                max_soft_score = np.max(current_label_soft_prediction)
-                factor = 255.0 / (max_soft_score - min_soft_score + 1e-12)
-                result_media_numpy = (factor * (current_label_soft_prediction - min_soft_score)).astype(np.uint8)
-
-                result_media = ResultMediaEntity(name=f'{label.name}',
-                                                 type='Soft Prediction',
-                                                 label=label,
-                                                 annotation_scene=dataset_item.annotation_scene,
-                                                 roi=dataset_item.roi,
-                                                 numpy=result_media_numpy)
-                dataset_item.append_metadata_item(result_media, model=self._task_environment.model)
+    def _add_predictions_to_dataset(self, prediction_results, dataset, dump_saliency_map):
+        """ Loop over dataset again to assign predictions. Convert from MMSegmentation format to OTE format. """
+        
+        for dataset_item, (prediction, feature_vector, saliency_map) in zip(dataset, prediction_results):
+            soft_prediction = np.transpose(prediction[0], axes=(1, 2, 0))
```

Comment:
In line 216, "ValueError: axes don't match array" error occurs while training.

Can you verify with this command? 
ote train ./external/model-preparation-algorithm/configs/segmentation/ocr-lite-hrnet-18-mod2/template.yaml \
--train-ann-files data/segmentation/custom/annotations/training \
--train-data-roots data/segmentation/custom/images/training \
--val-ann-files data/segmentation/custom/annotations/training \
--val-data-roots data/segmentation/custom/images/training \
--save-model-to data/seg-test \
params --learning_parameters.num_iters 2

Reply:
@chuneuny-emily The command works for me.
```
2022-07-28 11:05:45,385 | INFO : +-------+-------+-------+
|  aAcc | mDice |  mAcc |
+-------+-------+-------+
| 41.71 | 31.62 | 51.06 |
+-------+-------+-------+
2022-07-28 11:05:45,490 | INFO : Now best checkpoint is saved as best_mDice_epoch_2.pth.
2022-07-28 11:05:45,490 | INFO : Best mDice is 0.3162 at 2 epoch.
2022-07-28 11:05:45,490 | INFO : Exp name: /tmp/MPA-task-14nfowop/stage00_SegTrainer-train
2022-07-28 11:05:45,490 | INFO : Epoch(val) [2][4]	current_iters: 2, aAcc: 0.4171, mDice: 0.3162, mAcc: 0.5106, Dice.background: 0.0535, Dice.target: 0.5788, Acc.background: 0.0276, Acc.target: 0.9935
2022-07-28 11:05:46,493 | INFO : run task done.
2022-07-28 11:05:46,493 | INFO : Final model performance: Performance(score: 0.31620000000000004, dashboard: (11 metric groups))
2022-07-28 11:05:46,494 | INFO : called save_model: /tmp/MPA-task-14nfowop/stage00_SegTrainer-train/best_mDice_epoch_2.pth
2022-07-28 11:05:46,656 | INFO : train done.
2022-07-28 11:05:46,659 | INFO : infer()
2022-07-28 11:05:46,659 | INFO : initializing....
2022-07-28 11:05:46,659 | INFO : called _init_recipe()
2022-07-28 11:05:46,659 | INFO : train type = Incremental
2022-07-28 11:05:46,677 | INFO : initialized recipe = /home/yuchunli/git/training_extensions/external/model-preparation-algorithm/submodule/recipes/stages/segmentation/class_incr.py
2022-07-28 11:05:46,680 | INFO : initialized.
2022-07-28 11:05:46,680 | INFO : running task... kwargs = {'dump_features': True, 'dump_saliency_map': False}
2022-07-28 11:05:46,680 | INFO : called build_recipe()
2022-07-28 11:05:46,680 | INFO : seems to be passed stage yaml...
2022-07-28 11:05:46,680 | INFO : build_stage()
2022-07-28 11:05:46,681 | INFO : Training seed was set to 42 w/ deterministic=False.
2022-07-28 11:05:46,681 | INFO : work dir = /tmp/MPA-task-14nfowop/stage00_SegInferrer-train
2022-07-28 11:05:46,681 | INFO : CUDA_VISIBLE_DEVICES = None
2022-07-28 11:05:46,681 | INFO : configured logger at /tmp/MPA-task-14nfowop/stage00_SegInferrer-train with named 20220728_110546.log
2022-07-28 11:05:46,681 | INFO : configure!: training=False
Use load_from_local loader
2022-07-28 11:05:46,734 - mmseg - INFO - task config!!!!: training=False
Use load_from_local loader
2022-07-28 11:05:46,786 - mmseg - INFO - infer!
2022-07-28 11:05:46,787 - mmseg - INFO - Inferring on input source: data.test
2022-07-28 11:05:46,787 | INFO : test_mode : False
Use load_from_local loader
2022-07-28 11:05:46,922 - mmseg - INFO - ----------------- ClassIncrSegmentor.load_state_dict_pre_hook() called w/ prefix: 
2022-07-28 11:05:46,922 - mmseg - INFO - ['background', 'target'] -> ['background', 'target'] ([0, 1])
2022-07-28 11:05:47,276 | INFO : run task done.
2022-07-28 11:05:47,343 | INFO : called evaluate()
2022-07-28 11:05:47,343 | INFO : Computing mDice
function [__compute_dice_averaged_over_pixels] finished in 49.02982711791992 ms
2022-07-28 11:05:47,392 | INFO : mDice after evaluation: 0.5799746697654503
Performance(score: 0.5799746697654503, dashboard: (1 metric groups))
```

Can you check if your have updated the submodule with `git submodule update`?

Reply:
I used eugene/CVS-86547-enable-saliency-mpa-seg for both ote and submodule.